Sources/SwifterSwift/Combine/PublisherExtensions.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  PublisherExtensions.swift
+//  SwifterSwift
+//
+//  Created by Heecheon Park on 7/28/23.
+//  Copyright © 2023 SwifterSwift
+//
+
+#if canImport(Foundation) && canImport(Combine)
+import Foundation
+import Combine
+
+@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher {
+    /// SwifterSwift: Subscribes to four additional publishers and publishes a tuple upon receiving output from any of the publishers.
+    /// 
+    /// > Tip: The combined publisher doesn't produce elements until each of its upstream publishers publishes at least one element.
+    ///
+    ///     let pub = PassthroughSubject<Int, Never>()
+    ///     let pub2 = PassthroughSubject<Int, Never>()
+    ///     let pub3 = PassthroughSubject<Int, Never>()
+    ///     let pub4 = PassthroughSubject<Int, Never>()
+    ///     let pub5 = PassthroughSubject<Int, Never>()
+    ///     cancellable = pub
+    ///         .combineLatest(pub2, pub3, pub4, pub5)
+    ///         .sink { print("Result: \($0).") }
+    ///
+    ///     pub.send(1)
+    ///     pub.send(2)
+    ///     pub2.send(2)
+    ///     pub3.send(9)
+    ///     pub4.send(3)
+    ///     pub5.send(1)
+    ///
+    ///     pub.send(8)
+    ///     pub3.send(21)
+    ///
+    ///      Prints:
+    ///       Result: (2, 2, 9, 3, 1).
+    ///       Result: (8, 2, 9, 3, 1).
+    ///       Result: (8, 2, 21, 3, 1).
+    ///
+    /// - Parameters:
+    ///     - publisher1: A second publisher to combine with the first publisher.
+    ///     - publisher2: A third publisher to combine with the first publisher.
+    ///     - publisher3: A fourth publisher to combine with the first publisher.
+    ///     - publisher4: A fifth publisher to combine with the first publisher.
+    /// - Returns: A publisher that receives and combines elements from this publisher and four other publishers.
+    func combineLatest<P, Q, R, Y>(
```

Comment:
Why 4 specifically? I imagine that combine already has the ones 2 and 3 right?
Using [parameter packs](https://github.com/apple/swift-evolution/blob/main/proposals/0393-parameter-packs.md) wouldn't be a more generic way to handle this?

Reply:
Yes, Combine already provides "combineLatest" for 2 to 3 additional publishers and having additional 4th publisher comes in handy time to time.

I have tried to make reimplementation of combineLatest with parameter packs as you suggested but the feature is not mature enough to satisfy protocol constraint.

![스크린샷 2023-09-29 오전 9 10 48](https://github.com/SwifterSwift/SwifterSwift/assets/22357674/239692a6-b8ee-4dbc-a3c6-c9da2fca3257)

For example, each publisher in combineLatest must satisfy that the associated Failure must be identical to each iterating generic placeholder but currently Swift doesn't provide a way to put that constraint.